### PR TITLE
[backport to 2.7] Change documentation's `Show Source` link to GitHub

### PR DESCRIPTION
**Pull request by Mariatta**
Change documentation's `Show Source` link to GitHub (2.7 branch)
Render issue role as bpo-NNNN

### CODE REVIEW

File: Doc/tools/extensions/pyspecific.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def new_depart_literal_block(self, node):
 
 def issue_role(typ, rawtext, text, lineno, inliner, options={}, content=[]):
     issue = utils.unescape(text)
-    text = 'issue ' + issue
+    text = 'bpo- ' + issue
```
Comment (methane) [inline Doc/tools/extensions/pyspecific.py]: Is there space between 'bpo-' and issue number?

Comment (Mariatta) [reply Doc/tools/extensions/pyspecific.py]: No, there shouldn't be a space :sweat_smile:
Thanks for the review :) Will fix soon.

**Comment by methane** [comment]
Hm, the title of pull request is bit misleading.  doc of Python 2.7 doesn't include Misc/NEWS.
But this pull request backports these changes.

* Change link target of "**Source code:**" from hg.python.org to github
* issue role (`` :issue:`1234` ``) rendered as "bpo-1234" instead of "issue 1234" ([example](https://docs.python.org/3.7/whatsnew/3.7.html#other-language-changes))

**Review by methane — APPROVED** [review]
I did `make html` in local, and it works as expected.
LGTM.

**Comment by Mariatta** [comment]
Thanks @methane :) I updated the pull request description. 
What do you think?

**Comment by methane** [comment]
LGTM

**Pull request merged**